doc/timeline.md

```diff
@@ -0,0 +1,205 @@
+       , m_keyframes( value, 0_ra )
+       , m_currentValue( value )
+   {
+       // use the linear interpolator instead of the default one
+       m_keyframes.setInterpolator( Ra::Core::Animation::interpolate<Scalar> );
+   }
+
+   void initialize() override {}
+   void goTo( Scalar t ) {
+       // Here we use the value at time t.
+       m_currentValue = m_keyframes.at( t );
+   }
+   void update() {
+       // Here we can update other data that depend on the value.
+   }
+
+   KeyFrames m_keyframes;
+   Scalar m_currentValue;
+};
+
+struct MyTimeDependantSystem : public Ra::Engine::AbstractTimedSystem
+{
+   MyTimeDependantSystem( Ra::GuiBase::Timeline* timeline )
+       : Ra::Engine::AbstractTimedSystem()
+       , m_timeline( timeline )
+   {}
+
+   void generateTasks( Ra::Core::TaskQueue* taskQueue, const Ra::Engine::FrameInfo& ) override {
+       // Update all Components data.
+       for ( auto compEntry : m_components )
+       {
+           auto comp = static_cast<MyComponentWithKeyFrame*>( compEntry.second );
+           auto func = std::bind( &MyComponentWithKeyFrame::update, comp );
+           auto task = new Ra::Core::FunctionTask( func, "MyUpdateTask" );
+           taskQueue->registerTask( task );
+       }
+   }
+   void AnimationSystem::handleAssetLoading( Ra::Engine::Entity* entity, const Ra::Core::Asset::FileData* fileData ) {
+       // Add a Component whenever a file is loaded
+       auto comp = new MyComponentWithKeyFrame( "MyComponentWithKeyFrame_", entity );
+       registerComponent( entity, comp );
+       if ( m_timeline )
+       {
+           // on insertion we may want to get the frame value from elsewhere
+           auto inserter = [comp]( const Scalar& t ) {
+               // for example value(t) + 1
+               comp->m_keyframes.insertKeyFrame( t, comp->m_keyframes.at( t ) + 1_ra );
+           };
+           // No need for an update callback since the component owns the KeyFramedValue.
+           m_timeline->registerKeyFramedValue( comp, "KeyFrame", &comp->m_keyframes, inserter );
+       }
+   }
+   void goTo( Scalar t ) override {
+       // Update all Components' KeyFramedValue.
+       for ( auto compEntry : m_components )
+       {
+           compEntry.second->goTo( t );
+       }
+   }
+   void cacheFrame( const std::string& dir, uint frameID ) const override {}
+   bool restoreFrame( const std::string& dir, uint frameID ) override {}
+
+   Ra::GuiBase::Timeline* m_timeline{nullptr};
+};
+```
+
+## Dynamic KeyFramedValues:
+Those are `Ra::Core::Animation::KeyFramedValue`s that are not part of an `Ra::Engine::Entity`, `Ra::Engine::Component` or
+`Ra::Engine::RenderObject` data, but are used to keyframe some of its data.
+Dynamic `Ra::Core::Animation::KeyFramedValue`s must be created from and owned by the UI, and registered in the
+`Ra::GuiBase::Timeline`.
+They are usually bound to an UpdateCallback function since they have to update the
+object's data they are linked to.
+
```

Comment:
Here again, could you give a concrete example/use case anyone can understand ?
I don't see what option I should use (dynamic/static), despite the text above.

Reply:
I linked to 2 examples in the PR description (I could move them as official plugins and link the here):
  - the meshAnimator plugin which KeyFrames mesh vertices' position/normal though `value(t)+1`.
  - the cameraManipulation plugin extension which allows to keyframe the camera (position/rotation), but is not finished and lacks robustness.

Reply:
I added a plugin for entity animation in the github repo